zerver/tests/test_auth_backends.py

```diff
@@ -1891,6 +1891,65 @@ def test_validate_state(self) -> None:
             self.assertIn('login', result.url)
             m.assert_called_with("Sign in with Apple failed: bad state token.")
 
+class AppleAuthBackendNativeFlowTest(AppleIdAuthBackendTest):
```

Comment:
This should subclass SocialAuthBase because `AppleIdAuthBackendTest` has some tests that don't make sense to run in the native flow context

Reply:
The helper functions from AppleIdAuthBackendTest that we need here can be extracted into a mixin class.

Reply:
created a mixin class `AppleAuthMixin`